sdk/communication/Azure.Communication.Common/src/CommunicationIdentifier.cs

```diff
@@ -27,6 +27,31 @@ public abstract class CommunicationIdentifier : IEquatable<CommunicationIdentifi
         public override bool Equals(object obj)
             => obj is CommunicationIdentifier other && Equals(other);
 
+        /// <summary>
+        /// Overrides the equality operator.
+        /// </summary>
+        /// <param name="left">The first identifier to compare.</param>
+        /// <param name="right">The second identifier to compare.</param>
+        /// <returns>True if the types and <see cref="RawId"/> match.</returns>
+        public static bool operator ==(CommunicationIdentifier left, CommunicationIdentifier right)
+        {
+            return Equals(left, right) && left.GetType() == right.GetType();
+        }
+
+        /// <summary>
+        /// Overrides the non-equality operator.
+        /// </summary>
+        /// <param name="left">The first identifier to compare.</param>
+        /// <param name="right">The second identifier to compare.</param>
+        /// <returns>True if the types or <see cref="RawId"/> values are different.</returns>
+        public static bool operator !=(CommunicationIdentifier left, CommunicationIdentifier right)
+        {
+            return !Equals(left, right) || left.GetType() != right.GetType();
```

Comment:
I suspect that `GetType` is the cheaper check so with short-circuiting in mind I would change the order to 
```suggestion
            return left.GetType() != right.GetType() || !Equals(left, right);
```
Same for `==`

Reply:
I ran a quick benchmark, and it seems to be the case. well spotted! :)